src/iuse.cpp

```diff
@@ -7084,7 +7084,7 @@ int iuse::holster_gun(player *p, item *it, bool, const tripoint& )
         int minvol = maxvol / 3;
 
         auto filter = [maxvol, minvol]( const item &it ) {
-            return it.is_gun() && it.volume() <= maxvol && it.volume() >= minvol;
+            return it.is_gun() && it.volume() <= maxvol && it.volume() >= minvol && (it.skill() != "archery");
```

Comment:
This only filters things from the view, the player can still select items by their invlet (if it has one). That way one can *try* to put any item into the holster. It will currently tell you "That isn't a gun!"